src/firebase_functions/pubsub.py

```diff
@@ -117,8 +162,9 @@ def on_message_published(
     trigger = {} if pubsub_options is None else pubsub_options.metadata()
 
     def wrapper(func):
+
         @functools.wraps(func)
-        def pubsub_view_func(data: dict[str, Any]):
+        def pubsub_view_func(data: CloudEvent[Any]):
```

Comment:
is this not always a Cloud event with `MessagePublishedData`?

```suggestion
        def pubsub_view_func(data: CloudEvent[MessagePublishedData]):
```